utils/reviewsUtils.js

```diff
@@ -50,3 +51,36 @@ export async function postReview(review) {
     })
     return res.json();
 }
+
+async function getMovieReview(id) {
+  const res = await fetch(`${API_BASE}/reviews?filters[movie]=${id}`);
+  const payload = await res.json();
+  const modifiedArr = payload.data.map((obj) => ({
+    id: obj.id,
+    ...obj.attributes
+  }))
+  return modifiedArr;
+}
+
+export async function getAverageRating(id) {
+  const reviewsList = await getMovieReview(id);
+  const imdbRes =  await imdbRating(id);
```

Comment:
You are loading the rating from IMDB here, but you're only using it on line 78, in the `else` branch of the code. This means that you are loading this even if you don't need it, which will cause unnecessary delays.